sqlmesh/schedulers/airflow/client.py

```diff
@@ -198,6 +198,9 @@ def apply_plan(
         models_to_backfill: t.Optional[t.Set[str]] = None,
         end_bounded: bool = False,
         ensure_finalized_snapshots: bool = False,
+        directly_modified_snapshots: t.Optional[t.List[SnapshotId]] = None,
+        indirectly_modified_snapshots: t.Optional[t.List[SnapshotId]] = None,
```

Comment:
Will we ever care about which snapshot caused the indirectly modified one? We can consider making it:
```
t.Dict[str, t.List[SnapshotId]]
```
and use the snapshot name as a key.

Reply:
CC: @crericha

Reply:
for the specific thing im trying to do currently it isn't needed but i think this information would be nice to have in general so its probably worth including as a dict key

Reply:
Yeah, which snapshot caused the indirect mod is good info. In an ideal world, Airflow would provide a `Plan` object with a full `ContextDiff`.